tiny_hippo_run.py

```diff
@@ -38,41 +35,27 @@ def main():
     4. Sniffs packets on "wlan0" and analyzes the packet against signatures and privacy rules
     :return: nothing
     """
-    # 1) Pull and validate MAC addresses
-    pull_and_validate_addrs()
 
     # 2) Perform a system configuration security check
     for rule in rules_system_privacy:
         rule()
 
+    mac_addresses = DeviceInformation.get_mac_addresses()
+    # Note: Steps 2 and 3 happen simultaneously in the "sniff()" call, but are separated for clarity
+    # 2) Capture IoT packets only with crafted sniff
     # 3) Perform a scanning analysis of the IoT devices
-    ip_to_mac = __pair_ip_to_mac()
+    ip_to_mac = _pair_ip_to_mac(mac_addresses)
     for rule in rules_scanning_privacy:
         rule(ip_to_mac)
```

Comment:
This should be encapsulated in a Try/Except so the IDS doesn't crash. Similar to how it is done in packet_parse